### PR TITLE
Rest template: fixed jackson-dataformat-xml exclusion

### PR DESCRIPTION
**Pull request by irynamatveieva**
## Pull Request description

  - removed exclusion of jackson-dataformat-xml in pom.xml since it affects other places where needed
  - created util class to exclude jackson2xmlHttpMessageConverter for Rest Template
  - provided test for new class
  - excluded jackson2xmlHttpMessageConverter in all necessary places using util class method instead

PE PR: https://github.com/thingsboard/thingsboard-pe/pull/2724

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [ ] [Labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#about-labels) that classify your pull request have been added.
- [ ] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Crosslinks between PRs added. Required for internal contributors only.

## Back-End feature checklist

- [x] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [x] If new dependency was added: the dependency tree is checked for conflicts.
- [x] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [x] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.
- [x] If new yml property was added: make sure a description is added (above or near the property).

### CODE REVIEW

**Review by ShvaykaD — APPROVED** [review]

File: application/src/main/java/org/thingsboard/server/service/notification/channels/MicrosoftTeamsNotificationChannel.java [hunk 1]
```diff
@@ -108,7 +109,7 @@ public void sendNotification(MicrosoftTeamsNotificationTargetConfig targetConfig
                 message.setPotentialAction(List.of(actionCard));
             }
         }
-
+        RestTemplateUtil.excludeJackson2XmlHttpMessageConverter(restTemplate);
```
Comment (ShvaykaD) [inline application/src/main/java/org/thingsboard/server/service/notification/channels/MicrosoftTeamsNotificationChannel.java:112]: @irynamatveieva, please update this so we don't need to do this on each sendNotification request.

**Review by ShvaykaD — CHANGES_REQUESTED** [review]
Please see my comment above.

File: application/src/main/java/org/thingsboard/server/service/notification/channels/MicrosoftTeamsNotificationChannel.java [hunk 2]
```diff
@@ -43,16 +43,21 @@
...
-            .build();
+    private RestTemplate restTemplate;
+
+    public MicrosoftTeamsNotificationChannel(SystemSecurityService systemSecurityService) {
+        this.systemSecurityService = systemSecurityService;
+        this.restTemplate = new RestTemplateBuilder()
+                .setConnectTimeout(Duration.of(15, ChronoUnit.SECONDS))
+                .setReadTimeout(Duration.of(15, ChronoUnit.SECONDS))
+                .build();
+        RestTemplateUtil.excludeJackson2XmlHttpMessageConverter(this.restTemplate);
```
Comment (ViacheslavKlimov) [inline application/src/main/java/org/thingsboard/server/service/notification/channels/MicrosoftTeamsNotificationChannel.java:59]: Create a method for RestTemplate creation and use it everywhere:
`RestTemplateUtil.newRestTemplate()` which modifies the converters list under the hood and returns a builder.

File: application/src/main/java/org/thingsboard/server/service/security/auth/oauth2/CustomOAuth2ClientMapper.java [hunk 3]
```diff
@@ -57,6 +57,7 @@ private synchronized OAuth2User getOAuth2User(OAuth2AuthenticationToken token, S
         }
 
         RestTemplate restTemplate = restTemplateBuilder.build();
+        RestTemplateUtil.excludeJackson2XmlHttpMessageConverter(restTemplate);
```
Comment (ViacheslavKlimov) [inline application/src/main/java/org/thingsboard/server/service/security/auth/oauth2/CustomOAuth2ClientMapper.java:60]: I don't get why the RestTemplate is created on each method call

**Review by ViacheslavKlimov — CHANGES_REQUESTED** [review]

File: application/src/main/java/org/thingsboard/server/service/security/auth/oauth2/CustomOAuth2ClientMapper.java (hunk 3 above)
Comment (irynamatveieva) [reply application/src/main/java/org/thingsboard/server/service/security/auth/oauth2/CustomOAuth2ClientMapper.java:60]: see lines 52-57.

File: rest-client/src/main/java/org/thingsboard/rest/client/RestClient.java [hunk 4]
```diff
@@ -212,8 +213,8 @@ public RestClient(String baseURL) {
     }
 
     public RestClient(RestTemplate restTemplate, String baseURL) {
-        this.restTemplate = restTemplate;
-        this.loginRestTemplate = new RestTemplate(restTemplate.getRequestFactory());
+        this.restTemplate = newRestTemplate().configure(restTemplate);
```
Comment (ViacheslavKlimov) [inline rest-client/src/main/java/org/thingsboard/rest/client/RestClient.java:216]: Why not just `build`?

File: common/util/src/main/java/org/thingsboard/common/util/RestTemplateUtil.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.http.converter.xml.MappingJackson2XmlHttpMessageConverter;
+
+import java.util.List;
+
+public class RestTemplateUtil {
+
+    public static RestTemplateBuilder newRestTemplate() {
+        RestTemplateBuilder builder = new RestTemplateBuilder();
+        List<HttpMessageConverter<?>> messageConverters = builder.build().getMessageConverters();
```
Comment (ViacheslavKlimov) [inline common/util/src/main/java/org/thingsboard/common/util/RestTemplateUtil.java:28]: Extract message converters list to a constant.

**Pull request closed without merging**